changes/issue-2656-update-prometheus

```diff
@@ -0,0 +1,6 @@
+* Update to the latest version of the prometheus Go client library, resulting in some breaking changes in the exported metrics:
+    - `http_request_duration_microseconds` which was a Summary metric aggregated per quantile is now `http_request_duration_seconds_bucket`, an Histogram metric aggregated per bucket, with the buckets being `.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10`
+    - `http_request_duration_microseconds_sum` is now `http_request_duration_seconds_sum`
+    - `http_request_duration_microseconds_count` is now `http_request_duration_seconds_count`
+    - `http_request_size_bytes` which was a Summary metric aggregated per quantile is now `http_request_size_bytes_bucket`, an Histogram metric aggregated per bucket, with the buckets being `1KB, 100KB, 1MB, 100MB, 1GB`
+    - `http_response_size_bytes` which was a Summary metric aggregated per quantile is now `http_response_size_bytes_bucket`, an Histogram metric aggregated per bucket, with the buckets being `1KB, 100KB, 1MB, 100MB, 1GB`
```

Comment:
The breaking changes are basically imposed by the deprecation notice on the prometheus function we used to use: https://pkg.go.dev/github.com/prometheus/client_golang@v0.9.3/prometheus#InstrumentHandler

Notably: 

> The issues are the following: (1) It uses Summaries rather than Histograms. Summaries are not useful if aggregation across multiple instances is required. (2) It uses microseconds as unit, which is deprecated and should be replaced by seconds.

So this PR uses histograms instead of summaries and seconds instead of microseconds for the duration. THe other breaking change - using buckets instead of quantiles - also follows from the change from Summary to Histogram, see https://prometheus.io/docs/practices/histograms/#quantiles (here "client-side" means the side where metrics are observed, i.e. our fleet server):

> The essential difference between summaries and histograms is that summaries calculate streaming φ-quantiles on the client side and expose them directly, while histograms expose bucketed observation counts and the calculation of quantiles from the buckets of a histogram happens on the server side using the histogram_quantile() function.